### ISSUE TITLE
Problem with poor app file names

### ISSUE THREAD

**Issue by ChrisBlankDe** [SimonOfHH/D365BCAppHelper#10]
Microsoft is naming one of their apps "Microsoft_Tests for MX DIOT Extension..app". The Module has a problem with the two point before file extension.

Steps to reproduce
```powershell
$Artifact = Download-Artifacts -artifactUrl 'https://bcartifacts.azureedge.net/onprem/18.5.29545.0/us'
$App= Join-Path $Artifact "Applications\MX_DIOT\Test\Microsoft_Tests for MX DIOT Extension..app"
Get-D365BCManifestFromAppFile -Filename $App -Verbose
```

Repsonse:
```
VERBOSE: Getting information from c:\bcartifacts.cache\onprem\18.5.29545.0\us\Applications\MX_DIOT\Test\Microsoft_Tests for MX DIOT Extension..app
VERBOSE: Copying file c:\bcartifacts.cache\onprem\18.5.29545.0\us\Applications\MX_DIOT\Test\Microsoft_Tests for MX DIOT Extension..app to temporary directory
VERBOSE: Creating temporary path C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension.
VERBOSE: Copying C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension.\Microsoft_Tests for MX DIOT Extension..app to temporary path C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension. (as C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension.\Microsoft_Tests for MX DIOT Extension..app)
VERBOSE: Creating ZIP-file from APP-file
VERBOSE: New ZIP-file is located at C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension.\Microsoft_Tests for MX DIOT Extension..zip
VERBOSE: Extracting Microsoft_Tests for MX DIOT Extension..zip to C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension.\unzip
VERBOSE: Cleaning up / removing temporary path C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension.
Remove-Item : An object at the specified path C:\Users\me\AppData\Local\Temp\4b758a8a-703b-4f8b-be31-eb121925f632\Microsoft_Tests for MX DIOT Extension. does not exist.
At C:\Users\me\Documents\WindowsPowerShell\Modules\D365BCAppHelper\0.0.0.23\Export\Private\Get-D365BCManifestFromAppFile.ps1:265 char:17
+                 Remove-Item (Split-Path $Filename) -Force -Recurse
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Remove-Item], PSArgumentException
    + FullyQualifiedErrorId : Argument,Microsoft.PowerShell.Commands.RemoveItemCommand
```

https://github.com/SimonOfHH/D365BCAppHelper/blob/0a1b37fc2b7dafa163b397081083b80adae880f7/Export/Private/Get-D365BCManifestFromAppFile.ps1#L265

**Comment by SimonOfHH** [comment]
Hi Chris,

thanks for the info 👍 I'll check this asap and will include a fix. I'm currently working on an update anyway for new functions anyway, so your input is welcome 🤗

**Comment by SimonOfHH** [comment]
Hi @ChrisBlankDe 
I tried to reproduce this, but I can't 🤔 I used the exact 3 lines you provided to try it

```
VERBOSE: Getting information from c:\bcartifacts.cache\onprem\18.5.29545.0\us\Applications\MX_DIOT\Test\Microsoft_Tests for MX DIOT Extension..app
VERBOSE: Copying file c:\bcartifacts.cache\onprem\18.5.29545.0\us\Applications\MX_DIOT\Test\Microsoft_Tests for MX DIOT Extension..app to temporary directory
VERBOSE: Creating temporary path C:\Users\simon\AppData\Local\Temp\6c94fdbc-9370-4599-85ad-1857bd1f2258\Microsoft_Tests for MX DIOT Extension.
VERBOSE: Copying C:\Users\simon\AppData\Local\Temp\6c94fdbc-9370-4599-85ad-1857bd1f2258\Microsoft_Tests for MX DIOT Extension.\Microsoft_Tests for MX DIOT Extension..app to temporary path C:\Users\simon\AppData\Local\Temp\6c94fdbc-9370-4599-85ad-1857bd1f2258\Microsoft_Tests for MX DIOT Extension. (as C:\Users\simon\AppData\Local\Temp\6c94fdbc-9370-4599-85ad-1857bd1f2258\Microsoft_Tests for MX DIOT Extension.\Microsoft_Tests for MX DIOT Extension..app)
VERBOSE: Creating ZIP-file from APP-file
VERBOSE: New ZIP-file is located at C:\Users\simon\AppData\Local\Temp\6c94fdbc-9370-4599-85ad-1857bd1f2258\Microsoft_Tests for MX DIOT Extension.\Microsoft_Tests for MX DIOT Extension..zip
VERBOSE: Extracting Microsoft_Tests for MX DIOT Extension..zip to C:\Users\simon\AppData\Local\Temp\6c94fdbc-9370-4599-85ad-1857bd1f2258\Microsoft_Tests for MX DIOT Extension.\unzip
VERBOSE: Cleaning up / removing temporary path C:\Users\simon\AppData\Local\Temp\6c94fdbc-9370-4599-85ad-1857bd1f2258\Microsoft_Tests for MX DIOT Extension.


xmlns               : http://schemas.microsoft.com/navx/2015/manifest
App                 : App
IdRanges            : IdRanges
Dependencies        : Dependencies
InternalsVisibleTo  :
ScreenShots         :
SupportedLocales    : 
Features            :
PreprocessorSymbols :
SuppressWarnings    :
KeyVaultUrls        :
```

Any idea?

**Comment by ChrisBlankDe** [comment]
I just played around a bit and have to agree with you in part.
on my notebook with normal user -> error
on my notebook with admin -> works
on my buildserver with normal user -> error
on my buildserver with admin -> error

bevor testing i always run `$ErrorActionPreference ="Stop"`

Notebook is running Windows 11 Version 21H2 (22000.258) with PSVersion 5.1.22000.65 
Build Server is running Windows Server 2019 1809 (LTSC, 17763.2452) with PSVersion 5.1.17763.2268

**Comment by SimonOfHH** [comment]
Ok, I tested it again. And even though I can't reproduce it locally on my W10 machine, I think I know why the error happens. What's interesting is, that it doesn't happen for me under W10. 
When I try to manually create a folder with a name that ends with a dot (e.g. "test123.") Windows will automatically remove the dot at the end. So in the cleanup-part I modified the path from the file name a little bit (if it ends with a dot, remove the last character). I think I'll release an update later today - it would be great if you could verify it then. I'll ping as soon as I pushed the update.

**Comment by SimonOfHH** [comment]
Hi @ChrisBlankDe 
I merged the change (and some new functionality) and published the new version (0.0.0.24). Could you verify this with the new version?

**Comment by ChrisBlankDe** [comment]
Tested successfully on my local machine and one of our build servers.

THANKS
